sound/soc/sof/pcm.c

```diff
@@ -270,62 +272,18 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_PAUSE;
 		break;
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
-
-		/* check if the stream hw_params needs to be restored */
-		if (spcm->restore_stream[substream->stream]) {
-
-			/* restore hw_params */
-			ret = sof_restore_hw_params(substream, spcm, sdev);
-			if (ret < 0)
-				return ret;
-
-			/* unset restore_stream */
-			spcm->restore_stream[substream->stream] = 0;
-
-			/* trigger start */
-			stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
-		} else {
-
-			/* trigger pause release */
-			stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_RELEASE;
-		}
+		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_RELEASE;
 		break;
 	case SNDRV_PCM_TRIGGER_START:
-		/* fall through */
-	case SNDRV_PCM_TRIGGER_RESUME:
-
-		/* check if the stream hw_params needs to be restored */
-		if (spcm->restore_stream[substream->stream]) {
-
-			/* restore hw_params */
-			ret = sof_restore_hw_params(substream, spcm, sdev);
-			if (ret < 0)
-				return ret;
-
-			/* unset restore_stream */
-			spcm->restore_stream[substream->stream] = 0;
-		}
-
-		/* trigger stream */
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
-
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
-		/* fallthrough */
```

Comment:
Is this intended? I personally don't mind, I'm not advocating for such comments, but some maintainers / reviewers want them...

Reply:
@lyakh yes I intentionally removed it as there was comment saying that this might be superfluous.